### PR TITLE
add support for synthetic tests

### PR DESCRIPTION
**Pull request by tsiq-clemens**
Fixes https://github.com/testingisdocumenting/webtau/issues/765

Mark auto-generated `WebTauTest` as synthetic; that makes it more elegant to exclude them from metadata checks.

### CODE REVIEW

File: webtau-core/src/main/java/org/testingisdocumenting/webtau/reporter/WebTauTest.java [hunk 1]
```diff
@@ -17,17 +17,22 @@
 
 package org.testingisdocumenting.webtau.reporter;
 
-import org.testingisdocumenting.webtau.reporter.stacktrace.StackTraceCodeEntry;
-import org.testingisdocumenting.webtau.reporter.stacktrace.StackTraceUtils;
-import org.testingisdocumenting.webtau.time.Time;
-import org.testingisdocumenting.webtau.utils.FileUtils;
+import static org.testingisdocumenting.webtau.reporter.TestStatus.Errored;
```
Comment (tsiq-clemens) [inline webtau-core/src/main/java/org/testingisdocumenting/webtau/reporter/WebTauTest.java:20]: My IntelliJ's Google-style insisted on this import order:
1. import static
1. java
1. others

Many other files are clearly not following this ordering. Do we have a style guide to import that I overlooked?

File: webtau-feature-testing/examples/listeners/MetadataValidationTestListener.groovy [hunk 2]
```diff
@@ -6,7 +6,7 @@ import org.testingisdocumenting.webtau.reporter.WebTauTest
 class MetadataValidationTestListener implements TestListener {
     @Override
     void afterLastTestStatement(WebTauTest test) {
-        if (test.hasSteps() && !test.metadata.has('owner')) {
+        if (!test.isSynthetic() && !test.metadata.has('owner')) {
```
Comment (tsiq-clemens) [inline webtau-feature-testing/examples/listeners/MetadataValidationTestListener.groovy:9]: as @tsiq-nick mentioned, relying on steps is not future-proof

File: webtau-core/src/main/java/org/testingisdocumenting/webtau/reporter/WebTauTest.java (hunk 1 above)
Comment (MykolaGolubyev) [reply webtau-core/src/main/java/org/testingisdocumenting/webtau/reporter/WebTauTest.java:20]: we don't. I use whatever default in IDEA. I don't bother too much tbh.

File: webtau-feature-testing/examples/listeners/MetadataValidationTestListener.groovy (hunk 2 above)
Comment (MykolaGolubyev) [reply webtau-feature-testing/examples/listeners/MetadataValidationTestListener.groovy:9]: btw I am not using that account anymore, I use my personal one.

File: webtau-standalone-runner-groovy/src/test/groovy/org/testingisdocumenting/webtau/runner/standalone/StandaloneTestRunnerTest.groovy [hunk 3]
```diff
@@ -233,9 +235,10 @@ class StandaloneTestRunnerTest {
 
             runner.webTauTestList.size().should == 4
 
-            def beforeFirstTest = runner.webTauTestList.get(3)
-            beforeFirstTest.scenario.should == 'after all tests'
-            beforeFirstTest.shortContainerId.should == 'Teardown'
+            def afterAllTest = runner.webTauTestList.get(3)
```
Comment (MykolaGolubyev) [inline webtau-standalone-runner-groovy/src/test/groovy/org/testingisdocumenting/webtau/runner/standalone/StandaloneTestRunnerTest.groovy:238]: thanks for the renames

**Review by MykolaGolubyev — COMMENTED** [review]
Very nice! thank you

File: webtau-feature-testing/examples/listeners/MetadataValidationTestListener.groovy (hunk 2 above)
Comment (tsiq-clemens) [reply webtau-feature-testing/examples/listeners/MetadataValidationTestListener.groovy:9]: whoops, sorry

**Comment by MykolaGolubyev** [comment]
re failing build. I have a PR that is going to fix that, should be merged in soon

File: webtau-standalone-runner-groovy/src/main/groovy/org/testingisdocumenting/webtau/runner/standalone/StandaloneTest.groovy [hunk 4]
```diff
@@ -38,18 +38,22 @@ class StandaloneTest implements StepReporter {
...
         this.test.setScenario(description)
         this.test.setFilePath(workingDir.relativize(filePath))
         this.test.setShortContainerId(shortContainerId)
-        this.test.setSynthetic(isSyntheticTest)
 
         this.workingDir = workingDir
         this.code = code
     }
 
+    StandaloneTest asSynthetic() {
```
Comment (MykolaGolubyev) [inline webtau-standalone-runner-groovy/src/main/groovy/org/testingisdocumenting/webtau/runner/standalone/StandaloneTest.groovy:52]: 👍

**Comment by tsiq-clemens** [comment]
sure thing. I'll wait for your PR

**Pull request merged**